packages/ng/simple-select/src/lib/input/select-input.component.ts

```diff
@@ -0,0 +1,341 @@
+/* eslint-disable @angular-eslint/no-output-on-prefix */
+import { Overlay, OverlayConfig, OverlayContainer, OverlayModule, OverlayPositionBuilder, OverlayRef, ScrollStrategyOptions } from '@angular/cdk/overlay';
+import { Platform } from '@angular/cdk/platform';
+import { ComponentPortal } from '@angular/cdk/portal';
+import { CommonModule, DOCUMENT } from '@angular/common';
+import {
+	ChangeDetectionStrategy,
+	ChangeDetectorRef,
+	Component,
+	ComponentRef,
+	ElementRef,
+	EventEmitter,
+	forwardRef,
+	HostBinding,
+	HostListener,
+	Inject,
+	Injectable,
+	Injector,
+	Input,
+	OnDestroy,
+	OnInit,
+	Output,
+	TemplateRef,
+} from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { ReplaySubject, Subject, takeUntil } from 'rxjs';
+import { LuSelectPanelComponent, LuSelectPanelRef } from '../panel/index';
+import { ILuSelectPanelData, SELECT_ID, SELECT_LABEL, SELECT_LABEL_ID, SELECT_PANEL_DATA } from '../select.model';
+
+let selectId = 0;
+
+function selectIdFactory(): number {
+	return selectId++;
+}
+
+function selectLabelFactory(elementRef: ElementRef<HTMLElement>): HTMLLabelElement | undefined {
+	function getLabel(node: HTMLElement): HTMLLabelElement | undefined {
+		if (node instanceof HTMLLabelElement) {
+			return node;
+		}
+
+		if (!node.parentElement) {
+			return undefined;
+		}
+
+		return getLabel(node.parentElement);
+	}
+
+	return getLabel(elementRef.nativeElement);
+}
+
+function selectLabelIdFactory(label: HTMLLabelElement | undefined, selectId: number): string {
+	return label?.id || `lu-select-label-${selectId}`;
+}
+
+@Injectable()
+class LuSelectOverlayContainer extends OverlayContainer {
+	constructor(@Inject(DOCUMENT) document: Document, platform: Platform, @Inject(SELECT_LABEL_ID) private selectLabelId: string, @Inject(SELECT_ID) private selectId: number) {
+		super(document, platform);
+	}
+	protected override _createContainer(): void {
+		super._createContainer();
+		this._containerElement.setAttribute('aria-labelledby', this.selectLabelId);
+		this._containerElement.setAttribute('role', 'listbox');
+		this._containerElement.id = `lu-select-overlay-container-${this.selectId}`;
+	}
+}
+
+@Injectable()
+class LuSelectOverlay extends Overlay {}
+
+class SelectPanelRef<T> extends LuSelectPanelRef<T> {
```

Comment:
This file does a lot of things, maybe we could split it ? A file for each class ?

Reply:
It is definitely planned.  The whole panel code is going to be replaced with a more modular approach.

Reply:
But that will be done in a subsequent pr.